weblate/addons/views.py

```diff
@@ -35,63 +42,88 @@ def redirect_list(self, message=None):
 
     def get_context_data(self, **kwargs):
         result = super().get_context_data(**kwargs)
-        component = self.kwargs["component_obj"]
-        result["object"] = component
+        target = self.kwargs.get("component_obj") or self.kwargs.get("project_obj")
+        result["object"] = target
         installed = {x.addon.name for x in result["object_list"]}
-        result["available"] = sorted(
-            (
-                x
-                for x in ADDONS.values()
-                if x.can_install(component, self.request.user)
-                and (x.multiple or x.name not in installed)
-            ),
-            key=lambda x: x.name,
-        )
+
+        if isinstance(self.path_object, Component):
+            result["available"] = sorted(
+                (
+                    x
+                    for x in ADDONS.values()
+                    if x.can_install(target, self.request.user)
+                    and (x.multiple or x.name not in installed)
+                ),
+                key=lambda x: x.name,
+            )
+        elif isinstance(self.path_object, Project):
+            # Do I need to check permissions here for compnenets?
+            result["available"] = sorted(
+                (
+                    x
+                    for x in ADDONS.values()
+                    if x.project_scope and (x.multiple or x.name not in installed)
+                ),
+                key=lambda x: x.name,
+            )
```

Comment:
```suggestion
            result["available"] = sorted(
                (
                    x
                    for x in ADDONS.values()
                    if (x.multiple or x.name not in installed)
                ),
                key=lambda x: x.name,
            )
```

I think no filtering is needed here. The add-on execution should skip unsupported components, though. All `@receiver` in `weblate/addons/models.py` should get something like this in the loop:

```python
    if addon.project and not addon.addon.can_install(component):
       continue
```

I've made component variable available in all of these in 0df9af9b0963c7fc16f8c7a3244eeb23a3ff896c. This code that would deserve some refactoring as it is nearly the same for all the events, but let's not do it here – https://github.com/WeblateOrg/weblate/issues/10172